### PR TITLE
fix: Handle closing of session better (follow-up on #3)

### PR DESCRIPTION
**Pull request by FredeEB**
This solution dumbs down the logic for figuring out what buffers are tardis buffers, by just using the initial list of created buffers to also close them, as the initial list is essentially constant and only modified when closing all of the buffers anyways.

In #3 some discussion arose around how to properly end the session. I propose this solution

1. merge the initial 5 commits of #3 
2. merge this which addresse
3. create a new PR for the commits that @Nimjii has added on #3 as a separate PR

### CODE REVIEW

File: lua/tardis-nvim/init.lua [hunk 1]
```diff
@@ -18,27 +18,22 @@ local function get_git_root()
...
         command = 'git',
         args = { 'rev-parse', '--show-toplevel' },
-        on_exit = function (j, status)
-            if status ~= 0 then
-                error("git couldn't resolve the root", 4)
-            end
-
-            return j:result()
-        end
     }):sync()[1]
```
Comment (Nimjii) [inline lua/tardis-nvim/init.lua:21]: Won't this fail when the table is empty? I might be missing something.

File: lua/tardis-nvim/init.lua [hunk 2]
```diff
@@ -103,19 +94,17 @@ local function setup_keymap(origin, buffers)
...
-local function setup_autocmds(origin, buffers)
+local function setup_autocommands(buffers, origin)
     local group = vim.api.nvim_create_augroup('Tardis', {})
     for _, buffer in ipairs(buffers) do
         vim.api.nvim_create_autocmd({'BufUnload'}, {
             group = group,
             callback = function()
                 vim.api.nvim_del_augroup_by_id(group)
-                local current_buffer = vim.api.nvim_get_current_buf()
+                local to_close = vim.tbl_filter(function(other) return other.fd ~= buffer.fd end, buffers)
```
Comment (Nimjii) [inline lua/tardis-nvim/init.lua:104]: Considering that this is pretty similar to my updated solution (although cleaner, I will admit), I can just add this to my PR and then you can merge everything pretty easily. Unless there are still changes in my PR that you don't like.

File: lua/tardis-nvim/init.lua (hunk 1 above)
Comment (FredeEB) [reply lua/tardis-nvim/init.lua:21]: It returns `nil` from my testing. I've moved the error check to the `tardis` function further down

File: lua/tardis-nvim/init.lua (hunk 2 above)
Comment (FredeEB) [reply lua/tardis-nvim/init.lua:104]: That'd be possible too, however, to keep the history clean tidy I'd like to drop these two commits:
a937f85c8077895ce8beb196d5ba5b732de47c8e
bfa07a7e8ffdc42b8118b1168d601abb311e9a39. 

If you can pull those two out, I could merge yours, and then merge this afterwards

**Pull request closed without merging**